src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -237,6 +237,13 @@ public class StoreService : IStoreService
 
             switch (ex)
             {
+                case DataStoreException dataStoreException:
```

Comment:
I think you can skip the check inside:
```suggestion
                case DataStoreException dataStoreException when (dataStoreException.IsExternal):
                    throw;
```